cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -310,10 +315,22 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	if err := modeEnv.WriteTo(boot.InitramfsWritableDir); err != nil {
 		return err
 	}
-	// we need to put the file to disable cloud-init in the
-	// _writable_defaults dir for writable-paths(5) to install it properly
-	writableDefaultsDir := sysconfig.WritableDefaultsDir(boot.InitramfsWritableDir)
-	if err := sysconfig.DisableCloudInit(writableDefaultsDir); err != nil {
+
+	_, essSnaps, err := mst.ReadEssential("", []snap.Type{snap.TypeGadget})
+	if err != nil {
+		return err
+	}
+
+	// should only be one seed snap
+	gadgetSnap := squashfs.New(essSnaps[0].Path)
+
+	// we need to configure the recover system with defaults and such setup from
+	// the seed gadget
+	opts := &sysconfig.Options{
+		TargetRootDir: boot.InitramfsWritableDir,
+		GadgetSnap:    gadgetSnap,
```

Comment:
it's the default but probably AllowCloudInit: false,  would make it clearer what happens. same in any other place like this